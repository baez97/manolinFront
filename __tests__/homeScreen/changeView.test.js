import React from 'react';
import renderer from 'react-test-renderer';
import ChangeView from '../../components/homeComponents/changes/changeView';

jest.unmock("react-native");
jest.unmock("react");
jest.unmock("react-test-renderer");
jest.unmock("expo");

describe("ChangeView", () => {
    describe("Functions", () => {
        const mockedProps = {
            onPressFn : jest.fn()
        }

        const cV = new ChangeView();
        
        it("GetType returns Cambio when the type is change", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "change"
            };

            const expectedOutput = "Cambio";
            const actualOutput = cV.getType(mockedChange);
            expect( actualOutput ).toBe(expectedOutput);
        });

        it("GetType returns Librar when the type is free", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "free"
            };

            const expectedOutput = "Librar";
            const actualOutput = cV.getType(mockedChange);
            expect( actualOutput ).toBe(expectedOutput);
        });

        it("GetType returns an empty string when the type is not recognized", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "UNRECOGNIZED_TYPE"
            };

            const expectedOutput = "";
            const actualOutput = cV.getType(mockedChange);
            expect( actualOutput ).toBe(expectedOutput);
        });

        it("HandleLongPress works fine", () => {
            cV.handleLongPress();
            expect( mockedProps.handleLongPress ).toBeCalled();
        })
    });

    describe("View", () => {
        it("Matches the Snapshot when the type is 'change'", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "change"
            };

            const view         = <ChangeView change={mockedChange} />
            const rendered     = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });

        it("Matches the Snapshot when the type is 'free'", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "free"
            };

            const view         = <ChangeView change={mockedChange} />
            const rendered     = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });

        it("Matches the Snapshot when the type is invalid", () => {
            const mockedChange = {
                owner: "Ana",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "INVALID_TYPE"
            };

            const view         = <ChangeView change={mockedChange} />
            const rendered     = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });
    });
})