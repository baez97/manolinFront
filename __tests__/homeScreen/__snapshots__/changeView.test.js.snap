// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeView View Matches the Snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "elevation": 15,
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4287123141,
        4281607599,
      ]
    }
    endPoint={
      Array [
        0.8,
        1,
      ]
    }
    startPoint={
      Array [
        0.7,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 15,
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-between",
        "marginTop": 20,
        "padding": 20,
        "paddingLeft": 30,
        "paddingRight": 40,
        "width": 400,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 21,
            "textAlign": "left",
          }
        }
      >
        Lunes 5 de Febrero
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#cfbcf9",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 18,
              "marginTop": 8,
              "textAlign": "left",
            }
          }
        >
          Ana
        </Text>
        <Text
          style={
            Object {
              "color": "#cfbcf9",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 18,
              "marginTop": 8,
              "textAlign": "left",
            }
          }
        >
          Cambio
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 60,
          "marginTop": -5,
          "textAlign": "center",
        }
      }
    >
      T
    </Text>
  </ExponentLinearGradient>
</View>
`;
